source/common/microkernel_architecture/microkernel_modules/functionalities/AFunctionality.cpp

```diff
@@ -0,0 +1,84 @@
+#include <unistd.h>
+#include <iostream>
+#include "AFunctionality.hpp"
+
+//
+// Constructor and Destructor
+//
+AFunctionality::AFunctionality() : _thread(NULL), _running(false)
+{  
+}
+
+AFunctionality::~AFunctionality()
+{
+  stop();
+}
+
+//
+// Start the functionnality, threaded by default
+// and send a signal to FunctionalitiesManager class
+// to keep a trace of what functionalities are running right now
+//
+bool AFunctionality::start(bool threaded)
+{
+  if (_running)
```

Comment:
I don't know if it is GitHub but it seems the tabulations correspond to 2 spaces and not 4.